src/halmos/__main__.py

```diff
@@ -228,7 +229,15 @@ def rendered_log(log: EventLog) -> str:
     return f"{opcode_str}({args_str})"
 
 
-def render_trace(context: CallContext) -> None:
+def rendered_trace(context: CallContext) -> str:
+    output = io.StringIO()
+    render_trace(context, file=output)
+    result = output.getvalue()
+    output.close()
+    return result
```

Comment:
pretty neat! Suggest using a context manager here to make it even crispier

```suggestion
def rendered_trace(context: CallContext) -> str:
    with io.StringIO() as output:
        render_trace(context, file=output)
        return output.getvalue()
```